core/embed/trezorhal/flash.h

```diff
@@ -24,71 +24,41 @@
 #include <stdlib.h>
 #include "secbool.h"
 
-// see docs/memory.md for more information
-
-#if defined STM32F427xx || defined STM32F429xx
-#define FLASH_SECTOR_COUNT 24
-#elif defined STM32F405xx
-#define FLASH_SECTOR_COUNT 12
-#else
-#error Unsupported MCU
-#endif
+#define FLASH_OTP_NUM_BLOCKS 16
+#define FLASH_OTP_BLOCK_SIZE 32
 
-#define FLASH_SECTOR_BOARDLOADER_START 0
-//                                           1
-#define FLASH_SECTOR_BOARDLOADER_END 2
-
-//                                           3
-
-#if defined TREZOR_MODEL_T || defined TREZOR_MODEL_R || \
-    defined TREZOR_MODEL_DISC1
-#define FLASH_SECTOR_STORAGE_1 4
-#define FLASH_SECTOR_STORAGE_2 16
-#elif defined TREZOR_MODEL_1
-#define FLASH_SECTOR_STORAGE_1 2
-#define FLASH_SECTOR_STORAGE_2 3
-#else
-#error Unknown Trezor model
+#ifndef STM32U5A9xx
+#define FLASH_BYTE_ACCESS 1
```

Comment:
FTR, when writing this comment, I didn't realized that `FLASH_BYTE_ACCESS` is boolean defined/undefined, I thought it represented the number of bytes that have to be written in one access. Nevertheless, I think the comment remains more or less valid. When adding a new MCU it seems better to avoid defaults.

Reply:
instead of adding the ifdef here, i think it might be better to move this inside the concrete trezorhal - e.g. `trezorhal/stm32f4/platform.h` would have this defined, whereas `trezorhal/stm32u5/platform.h` would not. (or we can add there a new header `trezorhal/stm32f4/flash_specifics.h` or something). this wont change when changing inside MCU line i.e. to STM32U599xx, and when adding some other family like STM32L5 you need to check everything anyway

Reply:
moved  it here: https://github.com/trezor/trezor-firmware/commit/d0032acb4cafb4d044d3c920ecc832614e909c78